### ISSUE TITLE
bug: docker container refuses to start when there is an error in config.yml

### ISSUE THREAD

**Issue by BaccanoMob** [release-argus/Argus#450]
The logs when it happened where not useful since it did not specify where the error was.  For example, `panic: runtime error: invalid memory address or nil pointer dereference [signal sigsegv: segmentation violation code=0x1 addr=0xd8 pc=0x4b2740]` I got this for the most time.

Also, the [docs](https://release-argus.io/docs/config/defaults/) are outdated, the docs say notify and webhook falls under defaults but same config from [demo](https://release-argus.io/demo/config) shows its outside it as shown below. (Or my understanding with notify are wrong?)
```yaml
settings:
  log:
    level: DEBUG
defaults:
  service:
    latest_version:
      access_token: <secret>
    deployed_version:
      allow_invalid_certs: true
notify:
  default_gotify:
    type: gotify
    url_fields:
      host: gotify.example.io
      token: <secret>
  default_mattermost:
    type: mattermost
    url_fields:
      host: localhost
      port: "8081"
      token: <secret>
webhook:
  awx_more:
    type: github
    url: http://localhost:8081
    secret: <secret>
    silent_fails: true
  awx_other:
    type: github
    url: https://localhost:8081/api/v2/job_templates/36/github/
    secret: <secret>
    silent_fails: true
service:
  prometheus/alertmanager:
    latest_version:
      type: github
      url: prometheus/alertmanager
    notify:
      default_gotify: {}
      default_mattermost: {}
    webhook:
      awx:
        type: github
        url: https://localhost:8081/api/v2/job_templates/45/github/
        secret: <secret>
      awx_more: {}
    deployed_version:
      method: GET
      url: https://alert-manager.example.io/api/v2/status
      basic_auth:
        username: prometheus
        password: <secret>
      json: versionInfo.version
    dashboard:
      web_url: https://github.com/prometheus/alertmanager/blob/main/CHANGELOG.md
```

Honestly spent couple days trying to find the cause and since I couldn't even start the container I was fully lost on what to do till I realised I could actually start with a blank config.yml.

Best Solution: Better parser to give better error report. Though I believe it would be the most complicated to implement.

More feasible solutions: 

- When there is an error in config.yml it should, copy the current config.yml to the database folder and replace it was a blank config.yml (with settings set to debug maybe) and message in the UI stating there was some error in the config.yml and advise to use UI approach for a few services before editing config.yml again. 
- If the above is not possible, include "If the settings the config feels overwhelming, you can always start with a blank config.yml file and use the UI to add services, notify and webhooks." in this [section of the docs at the top.](https://release-argus.io/docs/config/) (Also, the docs states 6 parts while there is only 5?)

If you are cool with it, I can make the changes to the docs based on my experience with the config and set up when I can and create a pull request since I found a few mismatches and typos here and there as well.

**Comment by JosephKav** [comment]
`defaults.webhook` are defaults for all webhooks. You can then have a webhook that can be shared among multiple services by having it in `webhook.something`. `something` can then be used by any service, and will use any defaults in `defaults.webhook`

**Comment by JosephKav** [comment]
> panic: runtime error: invalid memory address or nil pointer dereference [signal sigsegv: segmentation violation code=0x1 addr=0xd8 pc=0x4b2740]

This would be an error in the code with pointer handling. The crash logs/config used would be needed to identify this issue. If the config is invalid, the container should log what's wrong

**Comment by BaccanoMob** [comment]
@JosephKav I went through the changes I made with the config.yml file. Below was one of the config.yml which caused that error. 

Config.yml
```yaml
settings:
  log:
    level: DEBUG

defaults:
  service:
    options:
      # interval: 2h
      interval: 10m
    latest_version:
      use_prerelease: true

# notify:
#   default_ntfy:
#     type: ntfy
#     url_fields:
#       Username: username
#       Password: password
#       Host: ntfy.sh
#       Topic: github
#     # params:
#     #   Actions: ''                        # see https://docs.ntfy.sh/publish/#action-buttons
#     #   Attach: https://example.com        # URL of an attachment
#     #   Cache: yes                         # Cache messages
#     #   Click: https://example.com         # URL to open when notification is clicked
#     #   Email: name@example                # Email address for email notifications
#     #   Filename: ''                       # Filename for attachment
#     #   Firebase: yes                      # Send to Firebase
#     #   Icon: https://example.com/icon.png # URL to an icon
#     #   Priority: default                  # Priority of the notification
#     #   Scheme: https                      # Server protocol
#     #   Tags: ''                           # Comma separated list of tags that may/may not map to emojis
#     #   Title: Argus                       # Title of the notification

# # generic+https://ntfy.$DOMAIN/topic?@authorization=Bearer+$TOKEN

# webhook:
#   default_ntfy2:
#     type: generic
#     url: https://ntfy.sh/topic?@authorization=Bearer+tk_somelong
#     # allow_invalid_certs: false

service:
  release-argus/argus:
    latest_version:
      type: github
      url: release-argus/argus
    dashboard:
      icon: https://raw.githubusercontent.com/release-argus/Argus/master/web/ui/react-app/public/favicon.svg
      icon_link-to: https://release-argus.io
      web_url: https://github.com/release-argus/Argus/blob/master/CHANGELOG.md

  amir20/dozzle:
    latest_version:
      type: github
      url: amir20/dozzle
    # notify:
    #   default_ntfy:
    #     params:
    #       Title: Dozzle Update
    # webhook:
    #   default_ntfy2:
    #     params:
    #       Title: Dozzle Update2

  # go-vikunja/api:
  #   latest_version:
  #     type: github
  #     url: go-vikunja/api
  #   notify:
  #     default_ntfy:
  #       params:
  #         Title: Vikunja Update
  #   dashboard:
  #     web_url: https://github.com/go-vikunja/api/blob/main/CHANGELOG.md
  #     icon: https://vikunja.io/images/vikunja.png
  #   webhook:
  #     default_ntfy2:
  #       params:
  #         Title: Vikunja Update2
```

Logs from dozzle (this repeats over and over again) when I ran with the above config. Settings is in DEBUG.
```
mem
load
Switching user UID:GID to 1000:1000
Applying perms to config.yml and argus.db
Starting...
DEBUG: 1/2 release-argus/argus Init
DEBUG: 2/2 amir20/dozzle Init
DEBUG: 3/2  Init
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd8 pc=0x4b2740]
all values are hidden
goroutine 1 [running]:
github.com/release-argus/Argus/service.(*Service).Init(0x79ac40?, 0x400009e450?, 0x0?, 0x0?, 0x3?, 0x40000ae208?, 0x40000ae3a0?, 0x40000ae330?, 0x40000ae210?)
	/build/service/init.go:50 +0x20
github.com/release-argus/Argus/config.(*Config).Init(0x40000ae008, 0x1)
	/build/config/init.go:60 +0x288
github.com/release-argus/Argus/config.(*Config).Load(0x40000ae008, {0x8477e7, 0xa}, 0x4000096130?, 0xfbdb20)
	/build/config/init.go:105 +0x658
main.main()
	/build/cmd/argus/main.go:51 +0x100
```

While, I was cleaning up the config file of comments to paste here, I ran it to get the latest logs. Same config without comments actually worked . Turns out the error comes with the comments for some reason. 

Since I was like confused with docs, I had made a lot of commented lines (pasted only half of it here) though I do not know how this causes an error.

config.yml without comments which works
```yaml
settings:
  log:
    level: DEBUG

defaults:
  service:
    options:
      interval: 10m
    latest_version:
      use_prerelease: true

service:
  release-argus/argus:
    latest_version:
      type: github
      url: release-argus/argus
    dashboard:
      icon: https://raw.githubusercontent.com/release-argus/Argus/master/web/ui/react-app/public/favicon.svg
      icon_link-to: https://release-argus.io
      web_url: https://github.com/release-argus/Argus/blob/master/CHANGELOG.md

  amir20/dozzle:
    latest_version:
      type: github
      url: amir20/dozzle
```

Logs of the above config.yml
```
Switching user UID:GID to 1000:1000
Applying perms to config.yml and argus.db
Starting...
DEBUG: 1/2 release-argus/argus Init
DEBUG: 2/2 amir20/dozzle Init
INFO: Found 2 services to monitor:
  - release-argus/argus
  - amir20/dozzle
VERBOSE: release-argus/argus, Tracking release-argus/argus at https://github.com/release-argus/argus every 10m
INFO: Listening on 0.0.0.0:8080/
VERBOSE: amir20/dozzle, Tracking amir20/dozzle at https://github.com/amir20/dozzle every 10m
VERBOSE: release-argus/argus, Potentially found new releases (new ETag "e3ba9bf791739601857baff44bf66a999a9eb7a05312293224b7bb6e220a76e2")
VERBOSE: amir20/dozzle, Potentially found new releases (new ETag "e613a9824977fb3ca300639c603081f06bcbd6432a1fb52838b955ba616f8bf5")
VERBOSE: release-argus/argus, Using cached releases (ETag unchanged)
INFO: release-argus/argus, Latest Release - "0.18.0"
DEBUG: db (data/argus.db), UPDATE status SET `latest_version` = ?,`latest_version_timestamp` = ? WHERE id = ?, [0.18.0 2024-10-04T17:43:44Z release-argus/argus]
DEBUG: db (data/argus.db), INSERT INTO status ('id', 'latest_version','latest_version_timestamp') VALUES (?,?,?), [release-argus/argus 0.18.0 2024-10-04T17:43:44Z]
DEBUG: db (data/argus.db), UPDATE status SET `deployed_version` = ?,`deployed_version_timestamp` = ? WHERE id = ?, [0.18.0 2024-10-04T17:43:44Z release-argus/argus]
VERBOSE: amir20/dozzle, Using cached releases (ETag unchanged)
INFO: amir20/dozzle, Latest Release - "8.5.5"
DEBUG: db (data/argus.db), UPDATE status SET `latest_version` = ?,`latest_version_timestamp` = ? WHERE id = ?, [8.5.5 2024-10-04T17:43:44Z amir20/dozzle]
DEBUG: db (data/argus.db), INSERT INTO status ('id', 'latest_version','latest_version_timestamp') VALUES (?,?,?), [amir20/dozzle 8.5.5 2024-10-04T17:43:44Z]
DEBUG: db (data/argus.db), UPDATE status SET `deployed_version` = ?,`deployed_version_timestamp` = ? WHERE id = ?, [8.5.5 2024-10-04T17:43:44Z amir20/dozzle]
```

Hope this helps to identify this issue!!

**Comment by JosephKav** [comment]
Thanks, that has helped me find the problem
```bash
DEBUG: 1/2 release-argus/argus Init
DEBUG: 2/2 amir20/dozzle Init
DEBUG: 3/2  Init
```
This showed it detected 3 service names, but that YAML only has 2.
We've got a function that gets the ordering of the services from the YAML as Go seems to choose its own orderings for reading/writing. The problem is that this wasn't handling comments correctly and so it recognised `  # go-vikunja/api:`, as a service, and unmarshalled the name of that service to be ''. Then the app tries to use Service[""], which is a nil pointer and causes that crash you sent.

Will do a PR shortly to fix this

```bash
❯ ./argus -config.file cmd/argus/config.yml
DEBUG: 1/2 release-argus/argus Init
DEBUG: 2/2 amir20/dozzle Init
INFO: Found 2 services to monitor:
  - release-argus/argus
  - amir20/dozzle
VERBOSE: release-argus/argus, Tracking release-argus/argus at https://github.com/release-argus/argus every 10m
INFO: Listening on 0.0.0.0:8080/
VERBOSE: release-argus/argus, Potentially found new releases (new ETag "e3ba9bf791739601857baff44bf66a999a9eb7a05312293224b7bb6e220a76e2")
VERBOSE: amir20/dozzle, Tracking amir20/dozzle at https://github.com/amir20/dozzle every 10m
VERBOSE: amir20/dozzle, Potentially found new releases (new ETag "e613a9824977fb3ca300639c603081f06bcbd6432a1fb52838b955ba616f8bf5")
```

**Comment by JosephKav** [comment]
Was failing to handle some comments in YAML

fixed in #455.
